admin/services/data-access/group.data.service.js

```diff
@@ -309,11 +309,29 @@ groupDataService.sqlMarkGroupAsDeleted = async (groupId) => {
  */
 groupDataService.sqlFindGroupsByIds = async (schoolId, groupIds) => {
   if (!schoolId || !groupIds || groupIds.length < 3) return false
-  const sql = `SELECT id, name 
+
+  let sql = `SELECT id, name 
     FROM ${sqlService.adminSchema}.[group] 
-    WHERE school_id=${schoolId} 
-    AND id IN (${groupIds})`
+    WHERE school_id=${schoolId}`
+
+  let whereClause = ' AND id IN ('
   const params = []
+  const ids = Object.values(groupIds)
+  for (let index = 0; index < ids.length; index++) {
+    if (ids[index].group_id) {
+      whereClause = whereClause + `@p${index}`
+      if (index < ids.length - 1) {
+        whereClause += ','
+      }
+      params.push({
+        name: `p${index}`,
+        value: ids[index].group_id,
+        type: TYPES.Int
+      })
+    }
+  }
+  whereClause = whereClause + ')'
```

Comment:
There is a where clause helper for this: sqlService.buildParameterList.  No need to maintain code in more than one place.

Reply:
@jon-shipley @GuyHarwood asked me to use 'sqlFindCheckWindowsAssignedToForms' as a reference to write this method, and so I did.

Reply:
...but if Jon has indicated that there is a reusable helper method that respects DRY, maybe take a look at it instead of creating yet more duplicate code???

Reply:
@GuyHarwood I've used it already, it's sqlService.buildParameterList

Reply:
yes, this comment is 3 days old.  see above where it says show/hide outdated.